api/oauth_endpoints.py

```
@@ -0,0 +1,271 @@
+                account = session.query(EmailAccount).filter(
+                    EmailAccount.email_address == account_email
+                ).first()
+
+                if account:
+                    # Update existing account with OAuth tokens
+                    account.customer_id = customer.id
+                    account.auth_method = "oauth"
+                    account.oauth_refresh_token = token_response["refresh_token"]
+                    account.oauth_access_token = token_response["access_token"]
+                    account.oauth_token_expires_at = datetime.utcnow() + \
+                        timedelta(seconds=token_response.get("expires_in", 3600))
```

Ruff (0.14.8)

Undefined name `timedelta`

(F821)